### ISSUE TITLE
Implement data fetching

### ISSUE THREAD

**Issue by guidsen** [guidsen/react-simple-universal#22]
What will be the best way to pre fetch data? In #1 we've talked a bit about it as well.
What options do we have and is most straightforward?

Extending the react-router `Route` component with a property that handles data fetching?
How would we handle data fetching from server and from client, because from the client we have to make AJAX requests where in the server we can return some kind of db promise with results.

cc @d6u @jackfranklin

**Comment by langpavel** [comment]
Nice isomorphic fetch is implemented in react-starter-kit.
They use **package.json** in subdir to separate client and server code like this:

``` json
{
  "private": true,
  "name": "fetch",
  "main": "./fetch.server.js",
  "browser": "./fetch.client.js"
}
```

See [react-starter-kit /src/core/fetch](https://github.com/kriasoft/react-starter-kit/tree/master/src/core/fetch)

**Comment by guidsen** [comment]
@langpavel If I'm right that means that you always make requests to an URL, right?

**Comment by langpavel** [comment]
Yes. It's one simplest and common pattern. This allows you to change backend in the future. Of course if you wish different code path on server and client you should use this trick on some API facade module.

**Comment by guidsen** [comment]
@langpavel Do you want to help building this data fetching in this repo? I see react-starter-kit is also using universal-router https://github.com/kriasoft/react-starter-kit/blob/master/src/server.js#L19.

If I'm correct this is just the functionality to make sure both the server will wait for the promise to resolve before rendering the first page.

**Comment by langpavel** [comment]
I'm not sure how much functionality must be done, I'll try to watch on this more deeply when I have enough time
